### ISSUE TITLE
vector/blas-subvector.h: build failure with GCC 15 (-Wtemplate-body)

### ISSUE THREAD

**Issue by orlitzky** [linbox-team/linbox#319]
GCC 15 is coming soon and we have started to build-test everything with it. This turned up a bug in linbox at https://bugs.gentoo.org/942337, which can be summarized as,

```
../../linbox/vector/blas-subvector.h: In constructor 'LinBox::BlasSubvector<_Vector>::BlasSubvector(Self_t&, size_t, size_t, size_t)':
../../linbox/vector/blas-subvector.h:121:20: error: 'LinBox::BlasSubvector<_Vector>::Self_t' has no member named 'data'; did you mean 'at'? [-Wtemplate-body[https://gcc.gnu.org/onlinedocs/gcc/C_002b_002b-Dialect-Options.html#index-Wno-template-body]]
  121 |             _ptr(V.data()+beg), _size(dim), _inc(inc), _field(&V.field()) {}
      |                    ^~~~
```

The new warning most likely arises because GCC now verifies templates when they are parsed, rather than only when they are instantiated.

**Comment by jgdumas** [comment]
Thank you. Should be solved by commit 4a1e139:
data() should have been getPointer() instead.
Also added a test in test-subvector.C.

**Comment by orlitzky** [comment]
Thank you!

**Comment by musicinmybrain** [comment]
> Thank you. Should be solved by commit [4a1e139](https://github.com/linbox-team/linbox/commit/4a1e1395804d4630ec556c61ba3f2cb67e140248): data() should have been getPointer() instead. Also added a test in test-subvector.C.

That fix works for me in Fedora. Thanks! I did run into an additional similar compiler error, https://github.com/linbox-team/linbox/issues/321.
